src/display/node_utils.js

```diff
@@ -27,56 +27,90 @@ if (typeof PDFJSDev !== "undefined" && PDFJSDev.test("MOZCENTRAL")) {
   );
 }
 
-let fs, canvas, path2d;
 if (isNodeJS) {
-  // Native packages.
-  fs = await __non_webpack_import__("fs");
-  // Optional, third-party, packages.
-  try {
-    canvas = await __non_webpack_import__("canvas");
-  } catch {}
-  try {
-    path2d = await __non_webpack_import__("path2d");
-  } catch {}
-}
+  // eslint-disable-next-line no-var
+  var packageCapability = Promise.withResolvers();
+  // eslint-disable-next-line no-var
+  var packageMap = null;
 
-if (typeof PDFJSDev !== "undefined" && !PDFJSDev.test("SKIP_BABEL")) {
-  (function checkDOMMatrix() {
-    if (globalThis.DOMMatrix || !isNodeJS) {
-      return;
-    }
-    const DOMMatrix = canvas?.DOMMatrix;
+  const loadPackages = async () => {
+    // Native packages.
+    const fs = await __non_webpack_import__("fs"),
+      http = await __non_webpack_import__("http"),
+      https = await __non_webpack_import__("https"),
+      url = await __non_webpack_import__("url");
 
-    if (DOMMatrix) {
-      globalThis.DOMMatrix = DOMMatrix;
-    } else {
-      warn("Cannot polyfill `DOMMatrix`, rendering may be broken.");
+    // Optional, third-party, packages.
+    let canvas, path2d;
+    if (typeof PDFJSDev !== "undefined" && !PDFJSDev.test("SKIP_BABEL")) {
```

Comment:
@Snuffleupagus Could you please elaborate on that line? This causes the code inside to be removed in generic builds and canvas to be always undefined in Node.js environments. Consequently, it's causing react-pdf unit tests to fail 🤔 Is that intentional?

Reply:
Most likely you're not using the *correct* build, given the browsers/environments that you want to support; please see https://github.com/mozilla/pdf.js/wiki/Frequently-Asked-Questions#faq-support

Note also that our CI tests run successfully in Node.js 18, when using a `legacy` build, so I don't think there's anything wrong with this line.